libsawtooth/src/journal/validation_rule_enforcer.rs

```diff
@@ -106,11 +106,11 @@ impl ValidationRuleEnforcer {
 }
 
 /// Parses the whole rules string, which is in the form "<rule1>;<rule2>;*"
-fn parse_rules(rules_str: &str) -> Result<Vec<Rule>, ValidationRuleEnforcerError> {
+fn parse_rules(rules_str: &str) -> Vec<Rule> {
     if rules_str.is_empty() {
-        Ok(vec![])
+        vec![]
     } else {
-        rules_str.split(';').map(Rule::from_str).collect()
+        rules_str.split(';').filter_map(Rule::parse).collect()
```

Comment:
Instead of writing a custom parse method, you can use `filter_map` and use just evaluate the error there. The FromStr implementation is could still be used in a CLI context, if we were to build a rule-specific subcommand.